controllers/volume/volume_controller.go

```diff
@@ -892,10 +907,37 @@ func sanitizeCNSErrorMessage(msg string) string {
 func attachmentToVolumeStatus(
 	volumeName string,
 	attachment cnsv1alpha1.CnsNodeVmAttachment) vmopv1.VirtualMachineVolumeStatus {
+
 	return vmopv1.VirtualMachineVolumeStatus{
 		Name:     volumeName, // Name of the volume as in the Spec
 		Attached: attachment.Status.Attached,
 		DiskUUID: attachment.Status.AttachmentMetadata[AttributeFirstClassDiskUUID],
 		Error:    sanitizeCNSErrorMessage(attachment.Status.Error),
+		Type:     vmopv1.VirtualMachineStorageDiskTypeManaged,
 	}
 }
+
+func updateVolumeStatusWithLimit(
+	ctx *pkgctx.VolumeContext,
+	c client.Reader,
+	status *vmopv1.VirtualMachineVolumeStatus) error {
+
+	var (
+		pvc    corev1.PersistentVolumeClaim
+		pvcKey = client.ObjectKey{Namespace: ctx.VM.Namespace, Name: status.Name}
```

Comment:
That `Name` isn't necessarily the PVC name - you want the `VirtualMachineVolume.PersistentVolumeClaim.ClaimName` value

Reply:
Doh, and you already told me as much. Sorry about that.